ci-operator/step-registry/cucushift/ota/postupgrade/cucushift-ota-postupgrade-commands.sh

```diff
@@ -159,6 +159,51 @@ function post-OCP-47197(){
     return 0
 }
 
+function post-OCP-53921(){
+    echo "Test Start: ${FUNCNAME[0]}"
+    local arch version
+    arch=$(oc get clusterversion version -ojson|jq -r '.status.conditions[]|select(.type == "ReleaseAccepted")|.message')
+    if [[ "${arch}" != *"Multi"* ]]; then
+        echo "The architecture info: ${arch} is not expected!"
+        return 1
+    fi
+    version="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    if [ -z "${version}" ] ; then
+        echo "Fail to get cluster version!"
+        return 1
+    fi
+    x_ver=$( echo "${version}" | cut -f1 -d. )
+    y_ver=$( echo "${version}" | cut -f2 -d. )
+    y_ver=$((y_ver+1))
+    ver="${x_ver}.${y_ver}"
+    if ! oc adm upgrade channel candidate-${ver}; then
+        echo "Fail to change channel to candidate-${ver}!"
+        return 1
+    fi
+    recommends=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates')
+    if [[ "${recommends}" == "null" ]]; then
+        echo "No recommended update available!"
+        return 1
+    fi
+    mapfile -t images < <(echo ${recommends}|jq -r '.[].image')
+    for image in ${images[*]}; do
+        if [ -z "${image}" ] ; then
```

Comment:
if `${images[*]}` is empty, the for loop will be skipped, right? Sounds need to check the length of `images` outside of the for loop?

Reply:
right, usually a build should include metadata, so i think if any issue came from the payload itself(build error?), we could ignore the metadata info check for this test.

Reply:
Hmm, my comment is not about image metadata checking, but about `image` is empty or not, if the `image` is an empty string, the for loop would directly skip it, would never come to this if condition.

Reply:
Got u now:) good point!

`availableUpdates` is a node like following(including image and version)
```
$ oc get clusterversion -o json|jq '.items[0].status.availableUpdates'
[
  {
    "image": "registry.ci.openshift.org/ocp/release@sha256:c7c3d2b39681c37817fb99c6f05140ee92a52920b7909d6e973d0025e6b60568",
    "version": "4.15.0-rc.1"
  }
]
```
there is a non-null check for availableupdate in https://github.com/openshift/release/pull/48934/files#diff-bcec571809d92fb0e609baf8907886244e300b0f2aa08759cc9f6cac2154a225R184, so if availableupdate is not null, images[] should not be null?

Reply:
If that, the line 190 - 193 sounds can be removed?

Reply:
Hmm, agree that in theory. `availableupdate!=null` is to ensure recommends list is not empty. But if there is one bad node join in the list unluckily, line 190 - 193 is for extra guarantee. 

In the extreme case that there is only one node in recommends list and it’s a bad node, then it will skip. I did not consider such a bad case, let me update it to handle it.

Reply:
> If that, images would be an empty array,

If there are more than one node in the array, then it's not an empty array but with a bad node with empty `image`. The loop will be skipped in the extreme case that there is only one bad node in the array.

Reply:
```
[
  {
    "image": "registry.ci.openshift.org/ocp/release@sha256:c7c3d2b39681c37817fb99c6f05140ee92a52920b7909d6e973d0025e6b60568",
    "version": "4.15.0-rc.1"
  },
 {
    "image": "",
    "version": "4.15.0-rc.2"
  },
 {
    "image": "registry.ci.openshift.org/ocp/release@sha256:xxxx",
    "version": ""
  },
]
```